### PR TITLE
Pivotal ID # 188114384: Scheduler SLURM Migration

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/188114384

- Create common Cluster config class
- Add option to execute scheduler jobs in SLURM
- Remove unused code

### CODE REVIEW

File: client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/LsfClusterClient.kt [hunk 1]
```diff
@@ -90,7 +90,6 @@ class LsfClusterClient(
     companion object {
         const val DONE_STATUS = "DONE"
         const val EXIT_STATUS = "EXIT"
-        const val PEND_STATUS = "PEND"
```
Comment (Juan-EBI) [inline client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/LsfClusterClient.kt:93]: why we have remove this one? (I remember seeing as a valid state)

**Review by Juan-EBI — APPROVED** [review]
LGTM

File: client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/LsfClusterClient.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/LsfClusterClient.kt:93]: It was never used. Maybe we need to do something with it

Comment (Juan-EBI) [reply client/cluster-client/src/main/kotlin/uk/ac/ebi/biostd/client/cluster/api/LsfClusterClient.kt:93]: yes is not used in a particular test but it is used when obtained job status using API

**Pull request merged**